### PR TITLE
cmake: additional message if Simde not found

### PR DESCRIPTION
**Pull request by fabinsch**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -89,7 +89,10 @@ add_project_dependency(
   "eigen3 >= 3.0.5"
 )
 if(BUILD_WITH_VECTORIZATION_SUPPORT)
-  add_project_dependency(Simde REQUIRED)
+  add_project_dependency(Simde)
+  if(NOT Simde_FOUND)
+    message(FATAL_ERROR "Set BUILD_WITH_VECTORIZATION_SUPPORT=OFF or install Simde.")
```
Comment (jcarpent) [inline CMakeLists.txt]: I would rather suggest defining a hint variable in the FindSimde.cmake file and fill it in the CMakeLists.txt.
In this way, we can get the benefit of both worlds

File: CMakeLists.txt [hunk 2]
```diff
@@ -89,7 +89,10 @@ add_project_dependency(
...
+  endif()
```
Comment (jcarpent) [inline CMakeLists.txt]:
```suggestion
  find_package(Simde QUIET)
  if(NOT Simde_FOUND)
    message(FATAL_ERROR "Set BUILD_WITH_VECTORIZATION_SUPPORT=OFF or install Simde on your system.\n If Simde is already installed, ensure that the CMake variable CMAKE_MODULE_PATH correctly points toward the location of FindSimde.cmake file.")
  endif()
  add_project_dependency(Simde REQUIRED)
```

Comment (jcarpent) [reply CMakeLists.txt]: This is another suggestion.

File: cmake-external/FindSimde.cmake [hunk 3]
```diff
@@ -1,10 +1,15 @@
 find_path(Simde_INCLUDE_DIR simde/simde-math.h)
 
+if(!${SIMDE_HINT_FAILURE})
+  set(SIMDE_HINT_FAILURE None)
+endif()
```
Comment (jcarpent) [inline cmake-external/FindSimde.cmake]:
```suggestion
if(NOT SIMDE_HINT_FAILURE)
  set(SIMDE_HINT_FAILURE None)
endif()
```

**Comment by fabinsch** [comment]
ok , I rebased and implement the version with a hint variable, leading to the following error message:

```
CMake Error at /home/fschramm/anaconda3/envs/master_pinocchio/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Simde (missing: Simde_INCLUDE_DIR)

      Reason given by package: Set BUILD_WITH_VECTORIZATION_SUPPORT=OFF or install Simde on your system.
   If Simde is already installed, ensure that the CMake variable CMAKE_MODULE_PATH correctly points toward the location of FindSimde.cmake file.

Call Stack (most recent call first):
  /home/fschramm/anaconda3/envs/master_pinocchio/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake-external/FindSimde.cmake:7 (find_package_handle_standard_args)
  cmake-module/package-config.cmake:110 (find_package)
  CMakeLists.txt:95 (add_project_dependency)
```

if `SIMDE_HINT_FAILURE` is not specified in the CMakeLists file you get the usual
```
CMake Error at /home/fschramm/anaconda3/envs/master_pinocchio/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Simde (missing: Simde_INCLUDE_DIR)
Call Stack (most recent call first):
  /home/fschramm/anaconda3/envs/master_pinocchio/share/cmake-3.23/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake-external/FindSimde.cmake:7 (find_package_handle_standard_args)
  cmake-module/package-config.cmake:110 (find_package)
  CMakeLists.txt:95 (add_project_dependency)
```

The version that you proposed @jcarpent leads to this output:
```
CMake Error at CMakeLists.txt:94 (message):
  Set BUILD_WITH_VECTORIZATION_SUPPORT=OFF or install Simde on your system.

   If Simde is already installed, ensure that the CMake variable CMAKE_MODULE_PATH correctly points toward the location of FindSimde.cmake file.
```

I think using the hint variable is the better solution, we add the possibility of additional information if people want for the FindSimde, but no one needs to use it. And its less changes in the CMakeLists.

**Comment by jcarpent** [comment]
Thanks for the quick report. I do agree with your conclusion.

**Pull request merged**